app/src/main/java/com/nmakademija/nmaakademija/fragment/NewsFragment.java

```diff
@@ -102,4 +105,42 @@ public void onClick(View view) {
 
         }
     }
+
+    @Override
+    public void onArticlesUpdated(ArrayList<Article> articles) {
+        List<Article> oldAcademics = adapter.articlesList;
+        if (oldAcademics.size() == articles.size()) {
+            for (int i = 0; i < articles.size(); i++) {
+                if (!oldAcademics.get(i).getTitle().equals(articles.get(i).getTitle()) ||
+                        !CompareStrings.equal(oldAcademics.get(i).getDescription(), articles.get(i).getDescription())) {
+                    newArticle(articles);
+                    return;
+                }
+            }
+            adapter.articlesList = articles;
+        } else {
+            newArticle(articles);
+        }
+    }
+
+    private void newArticle(final ArrayList<Article> articles) {
+        if (isVisible()) {
+            //noinspection ConstantConditions
+            Snackbar.make(getView(), "Naujiena", Snackbar.LENGTH_INDEFINITE)
```

Comment:
I don't think this is really needed. It's more than enough to show newest articles in the list.